### PR TITLE
Add category landing pages

### PR DESCRIPTION
**Pull request by darkwing**
Fixes https://github.com/mozilla/serviceworker-cookbook/issues/202

### CODE REVIEW

**Comment by digitarald** [comment]
Maybe a follow up bug, but could we have a description per category page for better SEO and giving users a primer on the technology.

File: parseRecipes.js [hunk 1]
```diff
@@ -9,9 +9,19 @@ var difficulties = {
   'Beginner': 1
 };
 
-var categoryOrder = ['General Usage', 'Offline', 'Beyond Offline', 'Performance', 'Web Push'];
+var categories = module.exports.categories = [
+  { title: 'General Usage', slug: 'general-usage' },
```
Comment (digitarald) [inline parseRecipes.js]: Should we include `service-workers` and `offline` for some SEO love?

Comment (darkwing) [inline parseRecipes.js]: I don't think we need it to be honest -- those keywords are everywhere in each demo.

File: parseRecipes.js [hunk 2]
```diff
@@ -9,9 +9,19 @@ var difficulties = {
...
+  { title: 'Offline', slug: 'offline' },
+  { title: 'Beyond Offline', slug: 'beyond-offline' },
+  { title: 'Performance', slug: 'performance' },
+  { title: 'Web Push', slug: 'web-push' }
+];
 
-module.exports = function(recipeSlugs) { 
+var categoryOrder = categories.map(function(category) {
+  return category.title;
+});
```
Comment (marco-c) [inline parseRecipes.js]: Could we order them as before? (simple to hard, then `performance` and then `web push`)

Comment (marco-c) [inline parseRecipes.js]: Ah, forget it, it's the same order.

File: parseRecipes.js (hunk 1 above)
Comment (digitarald) [inline parseRecipes.js]: In the demo but rarely in titles and headers so far. What about adding it to at least one section to see if it helps ranking.

File: gulpfile.js [hunk 3]
```diff
@@ -129,6 +130,27 @@ gulp.task('build:index', ['clean'], function buildIndex() {
...
+      return recipe.category === category.title;
+    }).map(function(recipe) {
+      return {
+        slug: recipe.slug,
+        title: recipe.name,
+        summary: marked(recipe.summary)
+      };
+    });
+
+    return renderFile('./src/tpl/category.html', {
```
Comment (digitarald) [inline gulpfile.js]: I am assuming that we are setting the page title? I would expect something along the lines of "Beyond Offline Recipes - ServiceWorker Cookbook" for categories and "API Analytics - Beyond Offline - Service Worker Cookbook" for recipes.

Comment (marco-c) [inline gulpfile.js]: We're not, I'm going to file an issue for that.

Comment (marco-c) [inline gulpfile.js]: Filed #226.

**Pull request merged**